### PR TITLE
chore: update resources for prometheus, document resource overrides

### PR DESCRIPTION
**Pull request by mjnagel**
## Description

Document added for resource/HA overrides across core packages.

Also ~doubles Prometheus' limits, but does not adjust the requests. This should ensure that Prometheus still schedules without requiring significant resources, but also allows it to consume more memory without hitting OOM errors.

## Related Issue

Related to https://github.com/defenseunicorns/uds-core/issues/551

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md) followed

### CODE REVIEW

File: docs/configuration/resource-configuration-and-ha.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
+---
+title: Resource Configuration and High Availability
+type: docs
+weight: 3.5
+---
+
+Depending on your environment setup and scale of your cluster and workloads you may need to modify components in UDS Core to run in a highly available manner or with modified resources. The below captures some of the common places resource overrides may be useful when deploying UDS Core.
```
Comment (chance-coleman) [inline docs/configuration/resource-configuration-and-ha.md]: ``` Depending on your environment and the scale of your cluster, you might need to adjust UDS Core components for high availability or to optimize resources. Below are common areas where resource overrides can be useful when deploying UDS Core. ```

File: docs/configuration/resource-configuration-and-ha.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+### Grafana
+
+To scale Grafana for high availability, its database must be externalized (see [Grafana's database configuration docs](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#database)). UDS Core has not yet done extensive testing on this setup. You can also override the `resources` helm value to customize Grafana pods' resource limits and requests (using the component and chart name of `grafana`).
+
+## Logging
+
+### Promtail
+
+By default Promtail runs as a daemonset, automatically scaling across all nodes to ensure logs are captured from each host. Typically Promtail does not need any other modifications, but you can customize its resource configuration similar by overriding the `resources` helm value (using the component and chart name of `promtail`).
```
Comment (chance-coleman) [inline docs/configuration/resource-configuration-and-ha.md]: nit:

``` similar by overriding the ```  -> ``` similar to overriding the ```

File: docs/configuration/resource-configuration-and-ha.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+              path: postgresql.username
+            - name: KEYCLOAK_DB_DATABASE
+              path: postgresql.database
+            - name: KEYCLOAK_DB_PASSWORD
+              path: postgresql.password
+```
+
+### AuthService
+
+AuthService can be configured in a highly-available mode with multiple replicas if an [external session store](https://docs.tetrate.io/istio-authservice/configuration/oidc#session-store-configuration) is provided (key value store like Redis/Valkey). For configuring an external session store you can set the `UDS_AUTHSERVICE_REDIS_URI` env when deploying or via your `uds-config.yaml`:
```
Comment (chance-coleman) [inline docs/configuration/resource-configuration-and-ha.md]: nit:

Authservice and Keycloak sections both start by saying the same thing in different words, IMO its easier to read/understand if using the same words.

File: docs/configuration/resource-configuration-and-ha.md (hunk 1 above)
Comment (mjnagel) [reply docs/configuration/resource-configuration-and-ha.md]:
```suggestion
Depending on your environment and the scale of your cluster, you might need to adjust UDS Core components for high availability or to optimize resources. Below are common areas where resource overrides can be useful when deploying UDS Core.
```

File: docs/configuration/resource-configuration-and-ha.md (hunk 2 above)
Comment (mjnagel) [reply docs/configuration/resource-configuration-and-ha.md]:
```suggestion
By default Promtail runs as a daemonset, automatically scaling across all nodes to ensure logs are captured from each host. Typically Promtail does not need any other modifications, but you can customize its resource configuration by overriding the `resources` helm value (using the component and chart name of `promtail`).
```

I think similar must've been me hitting my keyboard randomly - not sure it fits here 🤔

File: docs/configuration/resource-configuration-and-ha.md (hunk 3 above)
Comment (mjnagel) [reply docs/configuration/resource-configuration-and-ha.md]:
```suggestion
AuthService can be configured in a HA setup if an [external session store](https://docs.tetrate.io/istio-authservice/configuration/oidc#session-store-configuration) is provided (key value store like Redis/Valkey). For configuring an external session store you can set the `UDS_AUTHSERVICE_REDIS_URI` env when deploying or via your `uds-config.yaml`:
```

**Review by chance-coleman — APPROVED** [review]

**Pull request merged**